Sources/core/objects/Movable.h

```diff
@@ -34,7 +34,7 @@ class IMovable : public IOnMapObject
     void processMove();//for move
     virtual bool TryMove(Dir direct);
     bool CheckMoveTime();
-    bool CheckPassable();
+    virtual bool CheckPassable();
     bool MainMove();
```

Comment:
Yes, this is enough. But it is about overall game logic. 
Imagine we will make bow & arrows. It will be cool if arrows were heavily affected by air (wind). That will mean that projectile `Arrow` should have `passable_all_` as `Passable::AIR` (`SetPassable(D_ALL, Passable::AIR)`) . If projectiles will be deleted in `CheckPassable` it is only changes that will be needed. And I think it is cool.
So for future it will be better to delete projectiles in `CheckPassable`.